src/simple_pathfinding.h

```diff
@@ -0,0 +1,155 @@
+#ifndef PATHFINDIND_H
+#define PATHFINDIND_H
+
+#include "debug.h"
+#include "enums.h"
+
+#include <queue>
+#include <vector>
+
+namespace pf
+{
+
+struct node {
+    int x;
+    int y;
+    int dir;
+    int priority;
+
+    node( int x, int y, int dir, int priority = 0 ) :
+        x( x ),
```

Comment:
Preferably on one line, especially for trivial structs with few arguments. Not critical and worth delaying this PR on but worth considering in future.